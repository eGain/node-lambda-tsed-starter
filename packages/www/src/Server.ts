import "@tsed/platform-express" // /!\ keep this import
import "@tsed/ajv"
import "@tsed/swagger"

import * as controllers from "@project/controllers/index.js"
import { Configuration } from "@tsed/di"
import { PlatformLogMiddleware } from "@tsed/platform-log-middleware"
import { join } from "path"

import { config } from "./config/index.js"
import * as pages from "./controllers/pages/index.js"

@Configuration({
  ...config,
  acceptMimes: ["application/json"],
  httpPort: process.env.PORT || 8083,
  httpsPort: false, // CHANGE
  disableComponentsScan: true,
  ajv: {
    returnsCoercedValues: true
  },
  mount: {
    "/": [...Object.values(controllers)],
    "/pages": [...Object.values(pages)]
  },
  middlewares: [
    "cors",
    "cookie-parser",
    "compression",
    "method-override",
    "json-parser",
    { use: "urlencoded-parser", options: { extended: true } },
    PlatformLogMiddleware
  ],
  views: {
    root: join(process.cwd(), "../views"),
    extensions: {
      ejs: "ejs"
    }
  }
})
export class Server {}
