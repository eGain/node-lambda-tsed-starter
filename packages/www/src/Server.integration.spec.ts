import { PlatformTest } from "@tsed/platform-http/testing"
import SuperTest from "supertest"

import { Server } from "./Server.js"

describe("Server", () => {
  beforeEach(PlatformTest.bootstrap(Server))
  afterEach(PlatformTest.reset)

  it("should call GET /rest", async () => {
    const request = SuperTest(PlatformTest.callback())
    const response = await request.get("/rest").expect(404)

    expect(response.body).toEqual({
      errors: [],
      message: 'Resource "/rest" not found',
      name: "NOT_FOUND",
      status: 404
    })
  })

  it("should generate swagger.json", async () => {
    const request = SuperTest(PlatformTest.callback())
    const response = await request.get("/doc/swagger.json").expect(200)

    expect(response.body).toMatchSnapshot()
  })
})
