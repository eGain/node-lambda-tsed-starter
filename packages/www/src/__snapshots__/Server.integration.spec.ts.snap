// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Server > should generate swagger.json 1`] = `
{
  "components": {
    "schemas": {
      "GenericError": {
        "additionalProperties": true,
        "properties": {
          "message": {
            "description": "An error message",
            "minLength": 1,
            "type": "string",
          },
          "name": {
            "description": "The error name",
            "minLength": 1,
            "type": "string",
          },
        },
        "required": [
          "name",
          "message",
        ],
        "type": "object",
      },
      "GreetingRequest": {
        "properties": {
          "additionalMsg": {
            "type": "string",
          },
          "count": {
            "default": 2,
            "type": "number",
          },
          "name": {
            "example": "John",
            "minLength": 1,
            "type": "string",
          },
        },
        "required": [
          "name",
        ],
        "type": "object",
      },
      "NotFound": {
        "properties": {
          "errors": {
            "description": "A list of related errors",
            "items": {
              "$ref": "#/components/schemas/GenericError",
            },
            "type": "array",
          },
          "message": {
            "description": "An error message",
            "minLength": 1,
            "type": "string",
          },
          "name": {
            "default": "NOT_FOUND",
            "description": "The error name",
            "example": "NOT_FOUND",
            "minLength": 1,
            "type": "string",
          },
          "stack": {
            "description": "The stack trace (only in development mode)",
            "type": "string",
          },
          "status": {
            "default": 404,
            "description": "The status code of the exception",
            "example": 404,
            "type": "number",
          },
        },
        "required": [
          "name",
          "message",
          "status",
        ],
        "type": "object",
      },
      "Timeslot": {
        "properties": {
          "created_at": {
            "format": "date-time",
            "minLength": 1,
            "type": "string",
          },
          "description": {
            "default": "",
            "description": "Activity or Dresscode in HTML format",
            "maxLength": 500,
            "minLength": 3,
            "type": "string",
          },
          "end_date": {
            "format": "date-time",
            "minLength": 1,
            "type": "string",
          },
          "id": {
            "format": "uuid",
            "type": "string",
          },
          "label": {
            "example": "Hiking",
            "maxLength": 100,
            "minLength": 3,
            "type": "string",
          },
          "start_date": {
            "format": "date-time",
            "minLength": 1,
            "type": "string",
          },
          "updated_at": {
            "format": "date-time",
            "minLength": 1,
            "type": "string",
          },
        },
        "required": [
          "start_date",
          "end_date",
          "label",
          "description",
          "created_at",
          "updated_at",
        ],
        "type": "object",
      },
      "TimeslotCreate": {
        "properties": {
          "description": {
            "default": "",
            "description": "Activity or Dresscode in HTML format",
            "maxLength": 500,
            "minLength": 3,
            "type": "string",
          },
          "end_date": {
            "format": "date-time",
            "minLength": 1,
            "type": "string",
          },
          "id": {
            "format": "uuid",
            "type": "string",
          },
          "label": {
            "example": "Hiking",
            "maxLength": 100,
            "minLength": 3,
            "type": "string",
          },
          "start_date": {
            "format": "date-time",
            "minLength": 1,
            "type": "string",
          },
        },
        "required": [
          "start_date",
          "end_date",
          "label",
          "description",
        ],
        "type": "object",
      },
      "TimeslotUpdate": {
        "properties": {
          "description": {
            "default": "",
            "description": "Activity or Dresscode in HTML format",
            "maxLength": 500,
            "minLength": 3,
            "type": "string",
          },
          "end_date": {
            "format": "date-time",
            "minLength": 1,
            "type": "string",
          },
          "id": {
            "format": "uuid",
            "minLength": 1,
            "type": "string",
          },
          "label": {
            "example": "Hiking",
            "maxLength": 100,
            "minLength": 3,
            "type": "string",
          },
          "start_date": {
            "format": "date-time",
            "minLength": 1,
            "type": "string",
          },
        },
        "required": [
          "id",
          "start_date",
          "end_date",
          "label",
          "description",
        ],
        "type": "object",
      },
    },
  },
  "info": {
    "title": "Api documentation",
    "version": "1.0.0",
  },
  "openapi": "3.1.0",
  "paths": {
    "/hello": {
      "get": {
        "operationId": "helloControllerGet",
        "parameters": [
          {
            "in": "query",
            "name": "name",
            "required": false,
            "schema": {
              "type": "string",
            },
          },
        ],
        "responses": {
          "200": {
            "content": {
              "*/*": {
                "schema": {
                  "type": "string",
                },
              },
            },
            "description": "Success",
          },
        },
        "tags": [
          "HelloController",
        ],
      },
      "post": {
        "operationId": "helloControllerPost",
        "parameters": [],
        "requestBody": {
          "content": {
            "application/json": {
              "schema": {
                "$ref": "#/components/schemas/GreetingRequest",
              },
            },
          },
          "required": false,
        },
        "responses": {
          "200": {
            "content": {
              "application/json": {
                "schema": {
                  "type": "array",
                },
              },
            },
            "description": "Success",
          },
        },
        "tags": [
          "HelloController",
        ],
      },
    },
    "/timeslots": {
      "get": {
        "operationId": "timeslotsControllerGetTimeslots",
        "parameters": [],
        "responses": {
          "200": {
            "content": {
              "application/json": {
                "schema": {
                  "items": {
                    "$ref": "#/components/schemas/Timeslot",
                  },
                  "type": "array",
                },
              },
            },
            "description": "Success",
          },
        },
        "tags": [
          "TimeslotsController",
        ],
      },
      "post": {
        "operationId": "timeslotsControllerCreateTimeslot",
        "parameters": [],
        "requestBody": {
          "content": {
            "application/json": {
              "schema": {
                "$ref": "#/components/schemas/TimeslotCreate",
              },
            },
          },
          "required": false,
        },
        "responses": {
          "201": {
            "content": {
              "application/json": {
                "schema": {
                  "$ref": "#/components/schemas/Timeslot",
                },
              },
            },
            "description": "Created",
          },
        },
        "tags": [
          "TimeslotsController",
        ],
      },
    },
    "/timeslots/{id}": {
      "get": {
        "operationId": "timeslotsControllerGetTimeslotById",
        "parameters": [
          {
            "in": "path",
            "name": "id",
            "required": true,
            "schema": {
              "format": "uuid",
              "type": "string",
            },
          },
        ],
        "responses": {
          "200": {
            "content": {
              "application/json": {
                "schema": {
                  "$ref": "#/components/schemas/Timeslot",
                },
              },
            },
            "description": "Success",
          },
          "404": {
            "content": {
              "application/json": {
                "schema": {
                  "$ref": "#/components/schemas/NotFound",
                },
              },
            },
            "description": "Timeslot not found",
          },
        },
        "tags": [
          "TimeslotsController",
        ],
      },
      "put": {
        "operationId": "timeslotsControllerUpdateTimeslot",
        "parameters": [
          {
            "in": "path",
            "name": "id",
            "required": true,
            "schema": {
              "format": "uuid",
              "type": "string",
            },
          },
        ],
        "requestBody": {
          "content": {
            "application/json": {
              "schema": {
                "$ref": "#/components/schemas/TimeslotUpdate",
              },
            },
          },
          "required": false,
        },
        "responses": {
          "200": {
            "content": {
              "application/json": {
                "schema": {
                  "$ref": "#/components/schemas/Timeslot",
                },
              },
            },
            "description": "Success",
          },
          "404": {
            "content": {
              "application/json": {
                "schema": {
                  "$ref": "#/components/schemas/NotFound",
                },
              },
            },
            "description": "Timeslot not found",
          },
        },
        "tags": [
          "TimeslotsController",
        ],
      },
    },
  },
  "tags": [
    {
      "name": "HelloController",
    },
    {
      "name": "TimeslotsController",
    },
  ],
}
`;
